Sources/ApolloCore/CodegenCore/SelectionSet.swift

```diff
@@ -0,0 +1,67 @@
+public enum SelectionSetType<S: GraphQLSchema> {
+  case ObjectType(S.ObjectType)
+  case Interface(S.Interface)
+  case Union(S.Union)
+}
+
+public protocol SelectionSet: ResponseObject, Equatable {
+
+  associatedtype Schema: GraphQLSchema
+
+  /// The GraphQL type for the `SelectionSet`.
+  ///
+  /// This may be a concrete type (`ConcreteType`) or an abstract type (`Interface`).
+  static var __parentType: SelectionSetType<Schema> { get }
+}
+
+extension SelectionSet {
+
+  var __objectType: Schema.ObjectType { Schema.ObjectType(rawValue: __typename) ?? .unknownCase }
+
+  var __typename: String { data["__typename"] }
+
+  /// Verifies if a `SelectionSet` may be converted to a different `SelectionSet` and performs
+  /// the conversion.
+  ///
+  /// - Warning: This function is not supported for use outside of generated call sites.
```

Comment:
These comments don't wind up throwing a warning in Xcode, do they? (I don't think so but it'd be nice to make sure)

Reply:
No, they just add the "Warning" header in the documentation for that function

Reply:
Ooh - can you share a screenshot, I don't think I've seen that before!

Reply:
![image](https://user-images.githubusercontent.com/6243461/112693082-39467480-8e3d-11eb-9918-5b7d93ec43ea.png)

Reply:
neato